extension/icu/icu-datepart.cpp

```diff
@@ -291,11 +320,41 @@ struct ICUDatePart : public ICUDateFunc {
 		    });
 	}
 
+	struct BindStructData : public BindData {
+		using part_codes_t = vector<DatePartSpecifier>;
+		using bigints_t = vector<part_bigint_t>;
+		using doubles_t = vector<part_double_t>;
+
+		BindStructData(ClientContext &context, part_codes_t &part_codes_p, bigints_t &bigints_p, doubles_t &doubles_p)
+		    : BindData(context), part_codes(part_codes_p), bigints(bigints_p), doubles(doubles_p) {
+		}
+		BindStructData(const BindStructData &other)
+		    : BindData(other), part_codes(other.part_codes), bigints(other.bigints), doubles(other.doubles) {
+		}
+
+		part_codes_t part_codes;
+		bigints_t bigints;
+		doubles_t doubles;
+
+		bool Equals(const FunctionData &other_p) const override {
+			const auto &other = other_p.Cast<BindStructData>();
```

Comment:
The cast will throw if it's not the right type, can we check if the type matches and return false before doing this?

Reply:
This is a good point, but I looked at a _lot_ of other functions, and they all do this. Seems like something we should take a pass over to get them all. Right now, you can't even ask if the type is correct(!)